### ISSUE TITLE
503 Error at changelog publish on shared domainname Gitlab

### ISSUE THREAD

**Issue by ziouf** [python-semantic-release/python-semantic-release#363]
https://github.com/relekang/python-semantic-release/blob/83798b7bf94569014e00a5f030ac46a5d4992fff/semantic_release/hvcs.py#L363

You should use ${CI_SERVER_URL} instead of ${CI_SERVER_HOST} to build Gitlab.api_url()

You could implement like this:

```python
    @staticmethod
    def api_url() -> str:
        """Gitlab api_url property
        :return: The Gitlab instance API url
        """
        return config.get("hvcs_domain", os.environ.get("CI_SERVER_URL", "https://gitlab.com"))
```

This may fix the issue HTTP 503 issue at release posting phase when Gitlab server is behind a shared domainname with a url like this : https://<shared_domain>/gitlab

error : 
```
Posting changelog to HVCS
error: 503: <html><body><b>Http/1.1 Service Unavailable</b></body> </html>
ERROR: Job failed: exit code 1
```

**Comment by ziouf** [comment]
After some quick investigation, the use of CI_SERVER_HOST variable is not the root cause.

But the issue remains

**Comment by ziouf** [comment]
Looks related to https://github.com/relekang/python-semantic-release/issues/350#issuecomment-861024217

**Comment by seba-ban** [comment]
The same goes for [changelog](https://github.com/relekang/python-semantic-release/blob/master/semantic_release/changelog/changelog.py#L20).
Assuming https will produce incorrect links on http servers. As @ziouf mentioned, CI_SERVER_URL would be a better choice here. There's also a CI_SERVER_PROTOCOL variable in GitLab CI jobs.

**Comment by ziouf** [comment]
https://github.com/relekang/python-semantic-release/blob/83798b7bf94569014e00a5f030ac46a5d4992fff/semantic_release/vcs_helpers.py#L97-L110

I think it's better to use Gitlab-CI environement variables if available to get owner and project name values.

CI_PROJECT_NAMESPACE
CI_PROJECT_NAME

**Comment by codejedi365** [comment]
Resolved with #364
